From 2a352af1974b75dc1ab422eda230b06ddd768aed Mon Sep 17 00:00:00 2001
From: Frode Isaksen <frodex.isaksen@intel.com>
Date: Thu, 28 May 2015 11:46:56 +0200
Subject: [PATCH] HSU: do a HW reset when resuming

Do a HW reset when resuming, to avoid
a failure seen when GPS is in burst mode
and tracking. In this mode, the HSU executes
a suspend/resume cycle once per sec, and
after a certain time, no data is rx'ed
and GPS fails.
Symptom: IIR register is 0x00 after restoring
context.

Change-Id: I29c0c8c63af576651a31775bd5111cb5109100a5
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-37573
Signed-off-by: Frode Isaksen <frodex.isaksen@intel.com>
---
 drivers/tty/serial/hsu/hsu_core.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/tty/serial/hsu/hsu_core.c b/drivers/tty/serial/hsu/hsu_core.c
index f8b4bd81995a..a19ecafca01b 100644
--- a/drivers/tty/serial/hsu/hsu_core.c
+++ b/drivers/tty/serial/hsu/hsu_core.c
@@ -1506,6 +1506,9 @@ static void hsu_regs_context(struct uart_hsu_port *up, int op)
 	if (op == context_load) {
 		usleep_range(10, 100);
 
+		if (cfg->hw_reset)
+			cfg->hw_reset(up->port.membase);
+
 		serial_out(up, UART_LCR, up->lcr);
 		serial_out(up, UART_LCR, up->lcr | UART_LCR_DLAB);
 		serial_out(up, UART_DLL, up->dll);
-- 
1.9.1

